### PR TITLE
Add an overview page

### PR DESCRIPTION
**Pull request by feeblefakie**
## Description

This PR adds a new page describing the overview of ScalarDB since there are no docs doing that.

## Related issues and/or PRs

N/A

## Changes made

- Added descriptions about ScalarDB and its use cases.

## Checklist

> The following is a best-effort checklist. If any items in this checklist are not applicable to this PR or are dependent on other, unmerged PRs, please still mark the checkboxes after you have read and understood each item.

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes.
- [ ] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [ ] Tests (unit, integration, etc.) have been added for the changes.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

I will update README.md and index.md after this is merged to make them consistent with wording and refer to the docs site.

## Release notes

Added a page describing the overview of ScalarDB.

### CODE REVIEW

File: docs/overview.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# ScalarDB Overview
+
+## What is ScalarDB?
+
+ScalarDB is an HTAP engine for diverse databases. It achieves ACID transactions and real-time analytics across databases to simplify the complexity of managing multiple databases.  Existing solutions are designed to run only transactions or analytical queries across heterogeneous databases or both on homogeneous databases.
```
Comment (komamitsu) [inline docs/overview.md:5]:
```suggestion
ScalarDB is an HTAP engine for diverse databases. It achieves ACID transactions and real-time analytics across databases to simplify the complexity of managing multiple databases. Existing solutions are designed to run only transactions or analytical queries across heterogeneous databases or both on homogeneous databases.
```

File: docs/overview.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+### Manages consistency between multiple database
+Modern architectures (e.g., microservice architecture) encourage a system to separate a service and its database into smaller subsets to increase system modularity and development efficiency. However, managing separated (different kinds of) databases is challenging because applications must ensure those databases' correct states (i.e., consistencies). ScalarDB simplifies the management of such separated databases with a correctness guarantee (i.e., ACID with strict serializability), enabling users to focus on application development, not guaranteeing consistency between databases.
```
Comment (komamitsu) [inline docs/overview.md:19]:
```suggestion
Modern architectures (e.g., microservice architecture) encourage a system to separate a service and its database into smaller subsets to increase system modularity and development efficiency. However, managing separated (different kinds of) databases is challenging because applications must ensure those databases' correct states (i.e., consistencies). ScalarDB simplifies the management of such separated databases with a correctness guarantee (i.e., ACID with strict serializability), enabling users to focus on application development, guaranteeing consistency between databases.
```

File: docs/overview.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# ScalarDB Overview
+
+## What is ScalarDB?
```
Comment (komamitsu) [inline docs/overview.md:3]: What type of serializability and linearizability is supported in ScalarDB is mentioned in `Manages consistency between multiple database`, but I think it might be also good to mention it along with `ACID transactions` in this first section.

File: docs/overview.md (hunk 2 above)
Comment (feeblefakie) [reply docs/overview.md:19]: @komamitsu Thank you!
But, I think `not` is needed because this means that users don't need to guarantee consistency with ScalarDB.

Comment (komamitsu) [reply docs/overview.md:19]: Oh, I see. I misread the intention. It might sound to me a bit confusing...? But I'll rely on @josh-wong about that 😅

File: docs/overview.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+ScalarDB is an HTAP engine for diverse databases. It achieves ACID transactions and real-time analytics across databases to simplify the complexity of managing multiple databases. Existing solutions are designed to run only transactions or analytical queries across heterogeneous databases or both on homogeneous databases.
```
Comment (ochiai-kfj) [inline docs/overview.md:5]: Regarding the English expression, I will leave it to @jnmt 
I'd like to suggest constructing a story as a reader, shorten, common words, simple.
```suggestion
ScalarDB is a cross-database HTAP engine using a unique ACID transaction management.
There are several challenges in configuring an existing HTAP database.
- Transactions among different types of databases
- Trade-offs between performance and consistency
ScalarDB provides ACID transactions across multiple databases including non ACID compliant ones.
Analytical queries always answer consistent data without affecting performance.
This feature also relieves microservices developers from complicated designing tasks.
```
I just want to ask to re-arrange the composition, not the contents. Please replace my wrong contents into appropriate ones.

File: docs/overview.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+ScalarDB is a versatile solution that currently supports a wide range of databases. This includes major relational databases such as Oracle Database, PostgreSQL, MariaDB, MySQL, Microsoft SQL Server, SQLite, and their compatible databases (e.g., Amazon Aurora, Google AlloyDB, TiDB, YugabyteDB), as well as NoSQL databases like Amazon DynamoDB, Azure Cosmos DB, and Apache Cassandra.
+For details on which databases ScalarDB supports, refer to the [Supported Databases](docs/scalardb-supported-databases.md).
+
+## ScalarDB use cases
+
+ScalarDB can be used in various ways.
+Here are the three primary use cases of ScalarDB.
```
Comment (josh-wong) [inline docs/overview.md:13]:
```suggestion
ScalarDB can be used in various ways. This section describes the following three primary use cases for ScalarDB:

- Managing siloed databases easily
- Managing consistency between multiple database
- Reducing database migration hurdles
```

File: docs/overview.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+### Manages siloed databases simply
```
Comment (josh-wong) [inline docs/overview.md:15]:
```suggestion
### Managing siloed databases easily

```

File: docs/overview.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+Many enterprises comprise several organizations, departments, and business units to support agile business operations, which leads to siloed information systems; different organizations likely manage different applications with different databases. Managing such siloed databases is challenging because applications must communicate each database separately and properly deal with the differences between databases. ScalarDB simplifies the management of siloed databases with a unified interface, enabling users to treat the databases as if there were a single database.
+
+### Manages consistency between multiple database
```
Comment (josh-wong) [inline docs/overview.md:18]:
```suggestion
### Managing consistency between multiple database

```

File: docs/overview.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+Modern architectures (e.g., microservice architecture) encourage a system to separate a service and its database into smaller subsets to increase system modularity and development efficiency. However, managing separated (different kinds of) databases is challenging because applications must ensure those databases' correct states (i.e., consistencies). ScalarDB simplifies the management of such separated databases with a correctness guarantee (i.e., ACID with strict serializability), enabling users to focus on application development, not guaranteeing consistency between databases.
+
+### Lowers database migration hurdles
```
Comment (josh-wong) [inline docs/overview.md:21]:
```suggestion
### Reducing database migration hurdles

```

File: docs/overview.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+For details on which databases ScalarDB supports, refer to the [Supported Databases](docs/scalardb-supported-databases.md).
+
+## ScalarDB use cases
+
+ScalarDB can be used in various ways.
+Here are the three primary use cases of ScalarDB.
+
+### Manages siloed databases simply
+Many enterprises comprise several organizations, departments, and business units to support agile business operations, which leads to siloed information systems; different organizations likely manage different applications with different databases. Managing such siloed databases is challenging because applications must communicate each database separately and properly deal with the differences between databases. ScalarDB simplifies the management of siloed databases with a unified interface, enabling users to treat the databases as if there were a single database.
```
Comment (josh-wong) [inline docs/overview.md:16]: Revised for grammar and sentence flow. In particular, I re-worded the second sentence to avoid overusing the word `different`.

```suggestion
Many enterprises are comprised of several organizations, departments, and business units that support agile business operations, which often leads to siloed information systems. In addition, each organizations often manages their own applications by using different databases. Managing such siloed databases is challenging because applications must communicate with each database separately and properly deal with the differences between each database. 

ScalarDB simplifies managing siloed databases by providing a unified interface, enabling users to treat the databases as if they were a single database.
```

File: docs/overview.md (hunk 2 above)
Comment (josh-wong) [reply docs/overview.md:19]: Although I read this as the intended meaning, I could see how this could be misread. To reduce potential confusion, especially for when we translate this page, how about we change this paragraph as follows?:

```suggestion
Modern architectures, like a microservice architecture, encourage a system to separate a service and its database into smaller subsets to increase system modularity and development efficiency. However, managing disparate databases, especially of different kinds, is challenging because applications must ensure the correct states (or in other words, consistencies) of those databases.

ScalarDB simplifies managing such disparate databases with a correctness guarantee (or in other words, ACID with strict serializability), enabling you to focus on application development without worrying about guaranteeing consistency between databases.
```

File: docs/overview.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+
+### Manages consistency between multiple database
+Modern architectures (e.g., microservice architecture) encourage a system to separate a service and its database into smaller subsets to increase system modularity and development efficiency. However, managing separated (different kinds of) databases is challenging because applications must ensure those databases' correct states (i.e., consistencies). ScalarDB simplifies the management of such separated databases with a correctness guarantee (i.e., ACID with strict serializability), enabling users to focus on application development, not guaranteeing consistency between databases.
+
+### Lowers database migration hurdles
+Applications tend to be locked into a database by, for example, using database-specific capabilities. Such database lock-in discourages upgrading or changing the database because it requires rewriting the application. ScalarDB provides a unified interface over diverse databases; thus, once an application is written with ScalarDB, the application becomes portable, which helps achieve database migration without application rewrite.
```
Comment (josh-wong) [inline docs/overview.md:22]: I don't think `once an application is written with ScalarDB` is correct here, but I'm not totally sure if my suggestion is correct. Please confirm🙇🏻‍♂️

I also made a few changes to wording for clarity.

```suggestion
Applications tend to be locked into using a certain database because of the specific capabilities that the database provides. Such database lock-in discourages upgrading or changing the database because doing so requires re-writing the application.

ScalarDB provides a unified interface across diverse databases. Thus, once an application is written to include ScalarDB, the application becomes modular, which helps to achieve seamless database migration without needing to re-write the application.
```

**Review by josh-wong — CHANGES_REQUESTED** [review]
I've added some comments and suggestions. PTAL!

File: docs/overview.md (hunk 4 above)
Comment (jnmt) [reply docs/overview.md:5]: > Regarding the English expression, I will leave it to @jnmt

@ochiai-kfj Probably, Josh?? : )

Comment (jnmt) [inline docs/overview.md:5]: If we say an HTAP engine here, it might be good to add an analytical flavor in the use cases section since currently nothing related to analytics is mentioned there.

> to run only transactions or analytical queries across heterogeneous databases or both on homogeneous databases.

Also, this part is a bit hard to follow for me. I don't have a good idea, but something like the following? It's a little redundant, though...

Existing solutions are either of the following, although ScalarDB can run both transactional and analytical queries on heterogeneous databases.
- Run transactions only across heterogeneous databases
- Run analytical queries only across heterogeneous databases
- Run both transactional and analytical queries on homogeneous databases

**Review by jnmt — COMMENTED** [review]
Thanks for the good introduction. Basically, I like it. I just left a comment, but it's not a strong opinion. PTAL when you get a chance.

File: docs/overview.md (hunk 4 above)
Comment (feeblefakie) [reply docs/overview.md:5]: Thank you!
Good point. 
Let me update it based on your suggestions.

File: docs/overview.md (hunk 10 above)
Comment (feeblefakie) [reply docs/overview.md:24]: @josh-wong Thanks!
The latter paragraph sounds weird to me.
How about something like this?

```
ScalarDB provides a unified interface across diverse databases. Thus, once an application is written with the ScalarDB interface, the application becomes portable, which helps to achieve seamless database migration without needing to re-write the application.
```

File: docs/overview.md (hunk 9 above)
Comment (feeblefakie) [reply docs/overview.md:16]: @josh-wong Thanks.
Overall, looking good to me, but my Grammarly suggested this one.
How about this?
```suggestion
Many enterprises comprise several organizations, departments, and business units that support agile business operations, which often leads to siloed information systems. In addition, each organization often manages its own applications by using different databases. Managing such siloed databases is challenging because applications must communicate with each database separately and properly deal with the differences between each database. 
```

Comment (feeblefakie) [reply docs/overview.md:16]: @josh-wong 
1. `that support agile business operation` part sounds a little weird to me because I want to say the reasons for having several organizations.
2. `In addition, ` part also sounds a little different from my original intention because this sentence tried to describe more specific meaning of the first sentence.
3. My Grammarly suggests `are composed of` => `comprise` and `each organizations often manages their own applications` => `each organization often manages their its applications`. In both cases, the latter ones look better and correct.

What do you think?

File: docs/overview.md (hunk 4 above)
Comment (feeblefakie) [reply docs/overview.md:5]: @ochiai-kfj @josh-wong 
Thank you for the suggestions!
I got your ideas, but the suggested ones miss the analytics part (although it says HTAP), so it sounds a bit weird to me.
The suggestion is probably to use bullet points to clearly state the challenges, so let me update the paragraph based on that.

**Comment by feeblefakie** [comment]
@josh-wong @ochiai-kfj @jnmt @komamitsu 
I updated the PR based on your feedback.
The changes might not fully include your suggestions, but I tried my best to cover your intention from my perspective.
Please review it again!

File: docs/overview.md (hunk 3 above)
Comment (komamitsu) [reply docs/overview.md:5]: @feeblefakie Just remembered I left this. Do you think we don't need to mention "strict serializability" or something here? It might be a bit hard to mention it considering the analytical queries, but I think the strong correctness of ScalarDB for OLTP is one of the strong appeal points.

File: docs/overview.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+- NoSQL databases like Amazon DynamoDB, Apache Cassandra, and Azure Cosmos DB.
+
+For details on which databases ScalarDB supports, refer to [Supported Databases](scalardb-supported-databases.md).
+
+## ScalarDB use cases
+
+ScalarDB can be used in various ways. Here are the three primary use cases of ScalarDB.
+
+### Managing siloed databases easily
+Many enterprises comprise several organizations, departments, and business units to support agile business operations, which often leads to siloed information systems; different organizations likely manage different applications with different databases. Managing such siloed databases is challenging because applications must communicate each database separately and properly deal with the differences between databases.
```
Comment (josh-wong) [inline docs/overview.md:21]: Instead of a semicolon (`;`), how about we use `because` to connect these sentences?

```suggestion
Many enterprises comprise several organizations, departments, and business units to support agile business operations, which often leads to siloed information systems because different organizations likely manage different applications with different databases. Managing such siloed databases is challenging because applications must communicate with each database separately and properly deal with the differences between databases.
```

**Review by josh-wong — APPROVED** [review]
I've added one minor suggestion. Other than that, LGTM. Thank you!🙇🏻‍♂️

File: docs/overview.md (hunk 9 above)
Comment (josh-wong) [reply docs/overview.md]: > 1. `that support agile business operation` part sounds a little weird to me because I want to say the reasons for having several organizations.

Understood. I'm OK with leaving that part as is.

> 2. `In addition, ` part also sounds a little different from my original intention because this sentence tried to describe more specific meaning of the first sentence, not additional things.

Thank you for clarifying! I've made one minor suggestion in [another comment](https://github.com/scalar-labs/scalardb/pull/1621#discussion_r1558866268).

> 3. My Grammarly suggests `are composed of` => `comprise` and `each organizations often manages their own applications` => `each organization often manages its applications`. In both cases, the latter ones look better and correct.

I'm OK with using `comprise`. `are composed of` is passive voice and, arguably, maybe a bad habit of mine in cases like this😅 For the second one, that's a typo on my part.

File: docs/overview.md (hunk 10 above)
Comment (josh-wong) [reply docs/overview.md]: I'm OK with that. Thank you!

**Review by jnmt — APPROVED** [review]
LGTM! Thank you! I think Aki's suggestion regarding the comparison with existing solutions can be applied in a future improvement.

File: docs/overview.md [hunk 12]
```diff
@@ -0,0 +1,34 @@
+# ScalarDB Overview
+
+This page describes what ScalarDB is and its primary use cases.
+
+## What is ScalarDB?
+
+ScalarDB is a cross-database HTAP engine. It achieves ACID transactions and real-time analytics across diverse databases to simplify the complexity of managing multiple databases. Existing solutions are designed to run only transactions or analytical queries across heterogeneous databases, or they run transactions and analytical queries only on homogeneous databases.
```
[the remark this answers is not in the document]
Comment (feeblefakie) [reply docs/overview.md:7]: Good point.
Let me think about how I can improve the sentences.
But, let's first publish it and crystalize it later since something is better than nothing.

**Comment by feeblefakie** [comment]
@choplin 
Thank you for the comments! Good points. 
Let me think about how I can improve the sentences to make them easy to understand for normal engineers.
Let's first publish it and crystalize it later since showing something is better than nothing.

File: docs/overview.md (hunk 11 above)
Comment (feeblefakie) [reply docs/overview.md:21]: @josh-wong Thanks!
I used `;` to mean more like `specifically` to give concrete examples.
In that case, what would be a good choice?

**Comment by choplin** [comment]
@feeblefakie Sure. I agree to proceed in that way!

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

File: docs/overview.md (hunk 11 above)
Comment (josh-wong) [reply docs/overview.md:21]: I see. How about one of the following then?:

**Suggestion 1**

```suggestion
Many enterprises comprise several organizations, departments, and business units to support agile business operations, which often leads to siloed information systems; in particular, different organizations likely manage different applications with different databases. Managing such siloed databases is challenging because applications must communicate each database separately and properly deal with the differences between databases.
```

**Suggestion 2**

```suggestion
Many enterprises comprise several organizations, departments, and business units to support agile business operations, which often leads to siloed information systems. A prime example of this is how different organizations likely manage different applications with different databases. Managing such siloed databases is challenging because applications must communicate each database separately and properly deal with the differences between databases.
```

Comment (feeblefakie) [reply docs/overview.md]: @josh-wong I took the first suggestion. Thank you!

**Comment by feeblefakie** [comment]
I think the current one is good enough. Let me merge it first to publish it on the docs site. Then, I will create another PR to improve its content.

@josh-wong Please publish it.

**Comment by feeblefakie** [comment]
Created a new draft PR. I will work on it.
https://github.com/scalar-labs/scalardb/pull/1642
